internal/contentmanager/contentmanager.go

```
@@ -2,222 +2,143 @@ package contentmanager
-	// Doing this in a follow-up gives us the opportunity to verify that this functions
-	// as expected when wired up in the ClusterExtension reconciler before going too deep
-	// in optimizations.
-	i.mu.Lock()
-	if extCache, ok := i.extensionCaches[ce.GetName()]; ok {
-		extCache.Cancel()
-	}
-
-	cacheCtx, cancel := context.WithCancel(context.Background())
-	i.extensionCaches[ce.Name] = extensionCacheData{
-		Cache:  c,
-		Cancel: cancel,
+	dynamicSourcerer := &dynamicSourcerer{
+		// Due to the limitation outlined in the dynamic informer source,
+		// related to reusing an informer factory, we return a new informer
+		// factory every time to ensure we are not attempting to configure or
+		// start an already started informer
+		informerFactoryCreateFunc: func() dynamicinformer.DynamicSharedInformerFactory {
+			return dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicClient, time.Hour*10, metav1.NamespaceAll, func(lo *metav1.ListOptions) {
+				lo.LabelSelector = tgtLabels.AsSelector().String()
+			})
+		},
+		mapper: i.mapper,
 	}
-	i.mu.Unlock()
-
-	go func() {
-		err := c.Start(cacheCtx)
-		if err != nil {
-			i.Unwatch(ce)
-		}
-	}()
-
-	if !c.WaitForCacheSync(cacheCtx) {
-		i.Unwatch(ce)
-		return errors.New("cache could not sync, it has been stopped and removed")
-	}
-
-	return nil
+	cache = cmcache.NewCache(dynamicSourcerer, ce, i.syncTimeout)
+	i.caches[ce.Name] = cache
+	return cache, nil
 }
 
-// Unwatch will stop the cache for the provided ClusterExtension
-// stopping any watches on managed content
-func (i *instance) Unwatch(ce *v1alpha1.ClusterExtension) {
+// Delete stops and removes the Cache for the provided ClusterExtension
+func (i *managerImpl) Delete(ce *v1alpha1.ClusterExtension) {
 	if ce == nil {
 		return
 	}
 
 	i.mu.Lock()
-	if extCache, ok := i.extensionCaches[ce.GetName()]; ok {
-		extCache.Cancel()
-		delete(i.extensionCaches, ce.GetName())
+	defer i.mu.Unlock()
+	if cache, ok := i.caches[ce.Name]; ok {
+		cache.Close()
```

we should return the error that this returns (if it returns an error)

should we still delete it from the cache?

---

addressed in https://github.com/operator-framework/operator-controller/pull/1119/commits/af075c5137065a5e0b2e0cfc1b4debbb367f792f